graphql/schema.graphql

```diff
@@ -75,8 +76,8 @@ type Query {
   mainlineCommits(options: MainlineCommitsOptions!, buildVariantOptions: BuildVariantOptions): MainlineCommits
   taskNamesForBuildVariant(projectId: String!, buildVariant: String!): [String!]
   buildVariantsForTaskName(projectId: String!, taskName: String!): [BuildVariantTuple]
-  projectSettings(identifier: String!): ProjectSettings!
-  repoSettings(id: String!): RepoSettings!
+  projectSettings(identifier: String! @requireProjectAdmin): ProjectSettings!
```

Comment:
Anyone with view permission should be able to _see_ settings, edit access is only required really for the mutations.

Reply:
Oh shoot! I did not know this, I think there will be some front end implications as well. Would you like the "view only" access to be handled in this ticket or separately?

Reply:
I think it could make sense to do the directive as part of this ticket, and then any front-end facing implications as part of a different ticket, but whatever you prefer 💃 

For the legacy UI I think people with view only see the page normally but can't click save; I would also prefer that for the front end since the "disabled" view is such a drag (I have a memory of approving this with Lara back in the day also but as I am aged my memory may be faulty)

Reply:
Lol:

> as I am aged my memory may be faulty

That sounds good to me! Fwiw disabling all fields on the page would be fairly easy but I understand that that is not the most appealing view. And I'll add the view-only directive to this PR, thanks!